framework/wazuh/core/indexer/models.py

```diff
@@ -0,0 +1,93 @@
+import hashlib
+import os
+from dataclasses import InitVar, dataclass
+from datetime import datetime
+from hmac import compare_digest
+
+ITERATIONS = 100_000
+HASH_ALGO = 'sha256'
+
+
+def _generate_salt() -> bytes:
+    """Generate a random salt value.
+
+    Returns
+    -------
+    bytes
+        Random salt.
+    """
+    return os.urandom(16)
+
+
+def _hash_key(key: str, salt: bytes) -> str:
+    """Hash the given key using the provided salt.
+
+    Parameters
+    ----------
+    key : str
+        Value to hash.
+    salt : bytes
+        Value to use within derivation function.
+
+    Returns
+    -------
+    str
+        The hashed key.
+    """
+    return hashlib.pbkdf2_hmac(HASH_ALGO, key.encode('utf-8'), salt, ITERATIONS)
+
+
+@dataclass
+class Agent:
+    """Representation of a Wazuh Agent."""
+
+    id: str
+    name: str
+    key: str = None
+    groups: str = None
+    type: str = None
+    version: str = None
+    last_login: datetime = None
+    persistent_connection_node: str = None
+
+    raw_key: InitVar[str | None] = None
+
+    def __post_init__(self, raw_key: str | None):
+        if raw_key is not None:
+            self.key = self.hash_key(raw_key).decode('latin-1')
+
+    @staticmethod
+    def hash_key(raw_key: str) -> bytes:
```

Comment:
Could the hashing of the key be part of the `api/models/agent_added_model.py` code to have the real value of the key in the least time possible in memory? Or do you consider it necessary to have the hashing carried out in the indexer-related code?

Reply:
We must keep it here because it is a functionality to securely store in the indexer before that is not needed to handle a hash.